client/lib/user/user.d.ts

```diff
@@ -0,0 +1,66 @@
+/**
+ * External Dependencies
+ */
+import { EventEmitter } from 'events';
+
+export interface User extends EventEmitter {
+	initialize: () => Promise< void >;
+	clearStoreIfChanged: () => void;
+	get: () => UserData;
+	fetch: () => Promise< void > | boolean;
```

Comment:
it looks like this is always `Promise<void>` because if `this.fetching` is `false` it will create a new fetch Promise and return _that_

Reply:
there is some weird behavior,I'm looking at `client/lib/user/user.js:142` and it assigns `this.fetching` to a boolean `false` and also on line `:55`, it initializes this.isfetching to false... I don't know if it's possible to return fetching in one of those states or not... JavaScript...

Reply:
in this case, providing type declarations to _consumers_ of the API, we care most about describing what to expect. none of these types matter because nothing enforces them. in the JS code I didn't see anything that would return `this.fetching = false` and instead _only_ returns the Promise value. that's why I brought it up because I think the _intention_ is to only ever return a `Promise`

Reply:
ahh right because we check if fetching is truthy before returning it